azure_monitor/src/azure_monitor/protocol.py

```diff
@@ -19,20 +22,42 @@ def __repr__(self):
 
 
 class Data(BaseObject):
+    """Data
+
+    Args:
+        base_data: Data base data.
+        base_type: Data base type.
+    """
+
     __slots__ = ("base_data", "base_type")
 
     def __init__(self, base_data=None, base_type=None) -> None:
```

Comment:
Typing for `base_data` and `base_type` ?

Reply:
`base_data` would probably be an `Union` of accepted values, and `base_type` either a string or a `Literal` of accepted values.

Reply:
base_type is a string and base_data is any JSON parse able object (Custom events), would leave as "any" for now